src/bgw/job.c

```diff
@@ -603,63 +578,43 @@ ts_bgw_job_validate_job_owner(Oid owner, JobType type)
 		ReleaseSysCache(role_tup);
 		ereport(ERROR,
 				(errcode(ERRCODE_INVALID_AUTHORIZATION_SPECIFICATION),
-				 errmsg("permission denied to start %s background process as role \"%s\"",
-						job_type_names[type],
+				 errmsg("permission denied to start background process as role \"%s\"",
 						rform->rolname.data),
 				 errhint("Hypertable owner must have LOGIN permission to run background tasks.")));
 	}
 	ReleaseSysCache(role_tup);
 }
 
+static bool
+is_telemetry_job(BgwJob *job)
+{
+	return namestrcmp(&job->fd.proc_schema, INTERNAL_SCHEMA_NAME) == 0 &&
+		   namestrcmp(&job->fd.proc_name, "policy_telemetry") == 0;
+}
+
```

Comment:
Still find this an odd way to designate the telemetry job. Don't have a better suggestion though.